### PR TITLE
[TE] frontend - aaronucsd/Basic yaml editor with alert and subscripti…

### PR DESCRIPTION
**Pull request by aaronucsd**
…on group editors

1. Add new component for yaml
2. Add creation and edit features

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/yaml/template.hbs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<section class="te-page__top te-search-results {{if isEditModeActive "te-search-results--slim"}}">
+  <div class="container">
+    {{#self-serve-alert-yaml-details
+      detectionYaml=model.detectionYaml
+      isLoadError=isLoadError
+      displayMode="single"
+    }}
+
+      {{!-- {{#if (not isReplayPending)}}
```
Comment (jihaozh) [inline thirdeye/thirdeye-frontend/app/pods/manage/yaml/template.hbs]: Still need this?

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js [hunk 2]
```diff
@@ -880,89 +766,18 @@ export default Controller.extend({
...
-
-    },
 
     /**
      * Clears YAML content, disables 'save changes' button, and moves to form
      */
     cancelAlertYaml() {
-      set(this, 'disableYamlSave', true);
-      set(this, 'alertYamlContent', null);
+      //set(this, 'disableYamlSave', true);
```
Comment (jihaozh) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: clean these lines?

File: thirdeye/thirdeye-frontend/app/utils/constants.js [hunk 3]
```diff
@@ -139,3 +139,24 @@ rules:
...
+
+detectionConfigIds:
+ - 5773069
+
+fromAddress: thirdeye-dev@linkedin.com
+recipients:
+ to:
+  - "akrai@linkedin.com"
+ cc:
+  - "thirdeye-dev@linkedin.com"
```
Comment (akshayrai) [inline thirdeye/thirdeye-frontend/app/utils/constants.js]: This has and will potentially have a lot of LinkedIn specific things. I think we should move this out.

Maybe call an endpoint and let the backend return this value. LinkedIn stuff can be easily wrapped in the backend.

File: thirdeye/thirdeye-frontend/app/pods/components/yaml-editor/component.js [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+  isYamlParseable: true,
+  alertTitle: 'Define anomaly detection in YAML',
+  alertSettingsTitle: 'Define notification settings',
+  isEditMode: false,
+  showSettings: true,
+  disableYamlSave: true,
+  errorMsg: '',
+  alertYaml: null,           // The YAML for the anomaly alert detection
+  detectionSettingsYaml:  null,   // The YAML for the subscription group
+ //currentYamlSettings: null,// The YAML content for the notification settings
```
Comment (akshayrai) [inline thirdeye/thirdeye-frontend/app/pods/components/yaml-editor/component.js]: I'd prefer removing commented code if not required.

File: thirdeye/thirdeye-frontend/app/utils/constants.js (hunk 3 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/utils/constants.js]: this is the example template that shown as default in the alert Yaml edit page. To keep this simple, I think a yaml with dummy values (w/o linkedin specific information) would be enough for now.

Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/utils/constants.js]: this is the example template that shown as default in the alert Yaml edit page. To keep this simple, I think a yaml with dummy values (w/o linkedin specific information) would be enough for now.

Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/utils/constants.js]: this is the example template that shown as default in the alert Yaml edit page. To keep this simple, I think a yaml with dummy values (w/o linkedin specific information) would be enough for now.

Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/utils/constants.js]: Yep. Agreed. TOo much work to make it api driven for something so simple as a default. I can make it generic.

**Pull request merged**